rules_building_block/initial_access_github_new_user_agent_for_user.toml

```
@@ -0,0 +1,61 @@
+[metadata]
+creation_date = "2023/10/11"
+integration = ["github"]
+maturity = "production"
+min_stack_comments = "New fields added to GitHub Integration"
+min_stack_version = "8.8.0"
+updated_date = "2023/12/14"
+bypass_bbr_timing = true
+
+[rule]
+author = ["Elastic"]
+building_block_type = "default"
+description = """
+Detects a new user agent used for a GitHub user not previously seen in the last 14 days.
+"""
+from = "now-9m"
+index = ["logs-github.audit-*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "New User Agent For a GitHub User"
+risk_score = 21
+rule_id = "41761cd3-380f-4d4d-89f3-46d6853ee35d"
+severity = "low"
+tags = ["Domain: Cloud",
+        "Use Case: Threat Detection",
+        "Use Case: UEBA",        
+        "Tactic: Initial Access",
+        "Rule Type: BBR",
+        "Data Source: Github"
+        ]
+timestamp_override = "event.ingested"
+type = "new_terms"
+
+query = '''
+event.dataset:"github.audit" and event.category:"configuration" and
+github.user_agent:* and user.name:*
+'''
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1078"
+name = "Valid Accounts"
+reference = "https://attack.mitre.org/techniques/T1078/"
+[[rule.threat.technique.subtechnique]]
+id = "T1078.004"
+name = "Cloud Accounts"
+reference = "https://attack.mitre.org/techniques/T1078/004/"
+
+[rule.threat.tactic]
+id = "TA0001"
+name = "Initial Access"
+reference = "https://attack.mitre.org/tactics/TA0001/"
+
+[rule.new_terms]
+field = "new_terms_fields"
+value = ["github.user_agent", "user.name"]
```

```suggestion
value = ["user.name", "github.user_agent"]
```

We may want to swap these. `user.name` would be indexed where user agent is the cardinality so we can properly aggregate new user agents per user and not new users per user agent.

---

Does this work like that? I never knew. That is good to know

---

I didn't realize it worked this way either, thanks for clarifying